wagtail/wagtail
# Move all usage of versioned_static to media methods

To avoid staticfiles manifest file checking during loading of models, ensure that all versioned_static calls are only used inside media methods.

~Change versioned_static so that the `static()` call is done *after* the test for hashed staticfiles being used.~

Closes #5693.

To test:

```shell
pip install -e git+https://github.com/tomkins/wagtail.git@versioned-static#egg=wagtail
wagtail start versionedstaticdemo
cd versionedstaticdemo
echo "SECRET_KEY='secret'" > versionedstaticdemo/settings/local.py
cat > home/models.py <<EOF
from django.db import models

from wagtail.admin.edit_handlers import FieldPanel
from wagtail.core.models import Page
from wagtail.core.fields import RichTextField


class HomePage(Page):
    rich_text = RichTextField()

    content_panels = Page.content_panels + [
        FieldPanel("rich_text", classname="full"),
    ]
EOF
./manage.py makemigrations --settings=versionedstaticdemo.settings.production --dry-run --check
```
